wandb/sdk/interface/artifacts.py

```diff
@@ -165,23 +165,50 @@ def ref_url(self) -> str:
         raise NotImplementedError
 
 
-class ArtifactNotLoggedError(Exception):
-    """Raised for Artifact methods or attributes that require the logged artifact."""
+class ArtifactStatusError(AttributeError):
+    """Raised when an artifact is in an invalid state for the requested operation."""
 
     def __init__(
-        self, artifact: Optional["Artifact"] = None, attr: Optional[str] = None
+        self,
+        artifact: Optional["Artifact"] = None,
+        attr: Optional[str] = None,
+        msg: str = "Artifact is in an invalid state for the requested operation.",
     ):
-        desc = artifact.__class__.__name__ if artifact else "Artifact"
-        desc += f".{attr}" if attr else ""
-        super().__init__(
-            f"'{desc}' used prior to logging artifact or while in offline mode."
-            "Call wait() before accessing logged artifact properties."
-        )
+        object_name = artifact.__class__.__name__ if artifact else "Artifact"
+        method_id = f"{object_name}.{attr}" if attr else object_name
+        super().__init__(msg.format(artifact=artifact, attr=attr, method_id=method_id))
         # Follow the same pattern as AttributeError.
         self.obj = artifact
         self.name = attr
 
 
+class ArtifactNotLoggedError(ArtifactStatusError):
+    """Raised for Artifact methods or attributes only available after logging."""
+
+    def __init__(
+        self, artifact: Optional["Artifact"] = None, attr: Optional[str] = None
+    ):
+        super().__init__(
+            artifact,
+            attr,
+            "'{method_id}' used prior to logging artifact or while in offline mode. "
+            "Call wait() before accessing logged artifact properties.",
+        )
+
+
+class ArtifactFinalizedError(ArtifactStatusError):
+    """Raised for Artifact methods or attributes that can't be changed after logging."""
+
+    def __init__(
+        self, artifact: Optional["Artifact"] = None, attr: Optional[str] = None
+    ):
+        super().__init__(
+            artifact,
+            attr,
+            "'{method_id}' used on logged artifact. Can't add to finalized artifact.",
```

Comment:
Some questions from this error message:
- what is a finalized artifact? Is this a version that has already been logged and is thus immutable? (is this language we use elsewhere / does the user know what a finalized artifact is?)
- add what to a finalized artifact? Refers to adding new versions? Or adding new files to a version?
- see suggestion for rephrasing first sentence 

Sugestion: "'{method_id}' is used on an artifact that has already been logged. You cannot add ??? to this artifact as it has been finalized"